File: Arrays/Repeating And Missing Number/Better/code.py
def findNumbers(nums):
    n = len(nums)
    freq = [0]*(n+1)
    repeating, missing = -1, -1

    for x in nums:
        freq[x] += 1

    for i in range(1,n+1):
        if freq[i] == 0:
            missing = i
        if freq[i] > 1:
            repeating = i
    return repeating, missing

nums = [3,1,2,5,3]
print("Repeating:", findNumbers(nums)[0], "Missing:", findNumbers(nums)[1])

File: Arrays/Repeating And Missing Number/Brute/code.cpp
#include <bits/stdc++.h>
using namespace std;

pair<int,int> findMissingRepeating(vector<int>& nums) {
    int n = nums.size();
    int repeating = -1, missing = -1;
    for (int i = 1; i <= n; i++) {
        int count = 0;
        for (int j = 0; j < n; j++) {
            if (nums[j] == i) count++;
        }
        if (count == 2) repeating = i;
        else if (count == 0) missing = i;
    }
    return {repeating, missing};
}

int main() {
    vector<int> nums = {3,1,2,5,3};
    auto ans = findMissingRepeating(nums);
    cout << "Repeating: " << ans.first << " Missing: " << ans.second;
}
